### ISSUE TITLE
API token auth makes too many database calls

### ISSUE THREAD

**Issue by hansoncyu** [strapi/strapi#14512]
### Required System information

- Node.js version: 18.6.0
- NPM version: 8/13/2
- Strapi version: 4.4.0
- Database: Postgres
- Operating system: Mac

### Describe the bug
With the new API token system introduced in 4.4.0, every single API call now makes 10 db calls related to api tokens before even accessing the content data. For comparison, when using an older strapi version, there was only 1 api token related api call. Also all of these db calls are done serially because of awaited operations. This has more than doubled the average latency of API calls. I have not done any load testing but I would suspect that throughput would be noticeably reduced.

While this is technically not a correctness bug, I think that the significant degradation of performance and increased resource utilization leading to increased infrastructure costs make this issue worth some discussion.
Thanks in advance to the team.

### Steps to reproduce the behavior

1. Make a HTTP call to a strapi content type API with an API token.
2. Use a tracing tool to see how many database calls are performed.

### Expected behavior
Not perform 10+ database roundtrips for a single API call.

### Screenshots
This from a datadog dashboard that shows latency of our strapi service. The spike in latency is after I upgraded strapi to 4.4.0.
<img width="757" alt="Screen Shot 2022-09-30 at 12 34 42 PM" src="https://user-images.githubusercontent.com/28164678/193318736-622ea732-e26b-4201-bd12-aba049b9883d.png">

The below two images show before and after traces of a single API call. Each block corresponds to a postgres database call.
The before image has 7 calls: 1 is for api token and the other 6 are for content with relations.
The after image has 16 calls: 10 for api token and 6 for content.
This is an increase of 9 db roundtrips.
Before:
<img width="764" alt="Screen Shot 2022-09-30 at 12 48 11 PM" src="https://user-images.githubusercontent.com/28164678/193318791-71eaa9bf-5ba8-418f-8009-277c359bcf05.png">
After:
<img width="804" alt="Screen Shot 2022-09-30 at 12 48 22 PM" src="https://user-images.githubusercontent.com/28164678/193318808-b4fbe7ff-06d2-4730-a2d2-ea2a08e70f3b.png">

### Code snippets
There are two main causes for the increased number of db calls. The first is that this code now also populates the permissions field which is done as a separate call by the ORM instead of building a single query.
```
  const token = await strapi
    .query('admin::api-token')
    .findOne({ select: SELECT_FIELDS, populate: POPULATE_FIELDS, where: whereParams });
```
The other relevant change is [here](https://github.com/strapi/strapi/blob/main/packages/core/admin/server/strategies/api-token.js#L56). This new call to just update a last used field on the token causes many more db calls. The update code first queries for a token by id even though the auth code already has the token from an earlier query. Because of the permissions populate addition, this is an additional 2 db round trips. It then issues an db update for last used and another select on the token.

I am using a token with full access and unlimited expiration so I would expect my auth operations to be to just get a token and update the last used which is 2 calls.

Some ideas I have for fixing this performance bug are listed below. I am happy to discuss any other ideas that people come up with.
- Check the API token type first before querying the permissions table. A full access token doesn't need any db call.
- Allow the [update function](https://github.com/strapi/strapi/blob/main/packages/core/admin/server/services/api-token.js#L358) to take a token instead of doing another query for the token.
- The last used update call [here](https://github.com/strapi/strapi/blob/main/packages/core/admin/server/strategies/api-token.js#L56) probably doesn't need to be awaited. Let it execute in the background to not block the main request path.
- Updating a last used field doesn't need to trigger another select.
- I understand that the update function is pretty generic and can be used in different contexts but it might make sense to create another code path for auth requests so that updating a last used field can be much more efficient.

**Comment by innerdvations** [comment]
Thanks for the report! We'll have a fix for this as soon as possible.

**Comment by derrickmehaffy** [comment]
fixed by #14526
